discovery_engine_core/web-api/src/models.rs

```diff
@@ -0,0 +1,100 @@
+use chrono::{NaiveDateTime, Utc};
+use derive_more::Display;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use xayn_discovery_engine_ai::{Document as AiDocument, DocumentId, Embedding};
+
+/// Unique identifier of the [`Document`].
+#[derive(Debug, PartialEq, Eq, Clone, Copy, Hash, Serialize, Deserialize, Display)]
+#[repr(transparent)]
+#[cfg_attr(test, derive(Default))]
+pub struct Id(Uuid);
```

Comment:
In `core` crate we also do that, I think it's because `DocumentId` doesn't derive `Serialize` & `Deserialize`

Reply:
In `core` we did it because was a new project and we started using bits of xayn_ai later. I think here we should reuse the abstraction from core or ai when possible. We can add the derive to them if necessary.